tests/models/test_dataset.py

```diff
@@ -0,0 +1,121 @@
+# ***** BEGIN GPL LICENSE BLOCK *****
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# The Original Code is Copyright (C) 2021, TIALab, University of Warwick
+# All rights reserved.
+# ***** END GPL LICENSE BLOCK *****
+"""Tests for predefined dataset within toolbox."""
+
+import os
+import shutil
+
+import numpy as np
+import pytest
+
+from tiatoolbox import rcParam
+from tiatoolbox.models.dataset import DatasetInfoABC, KatherPatchDataset, PatchDataset
+from tiatoolbox.utils.misc import download_data, unzip_data
+
+
+def test_DatasetABC():
+    """Test for ABC."""
+    # test defining a subclass of dataset info but not defining
+    # enforcing attributes - should crash
+    with pytest.raises(TypeError):
+
+        # intentionally created to check error
+        # skipcq
+        class Proto(DatasetInfoABC):
+            def __init__(self):
+                self.a = "a"
+
+        # intentionally created to check error
+        Proto()  # skipcq
+    with pytest.raises(TypeError):
+
+        # intentionally created to check error
+        # skipcq
+        class Proto(DatasetInfoABC):
+            def __init__(self):
+                self.inputs = "a"
+
+        # intentionally created to check error
+        Proto()  # skipcq
+    with pytest.raises(TypeError):
+        # intentionally created to check error
+        # skipcq
+        class Proto(DatasetInfoABC):
+            def __init__(self):
+                self.inputs = "a"
+                self.labels = "a"
+
+        # intentionally created to check error
+        Proto()  # skipcq
+    with pytest.raises(TypeError):
+
+        # intentionally created to check error
+        # skipcq
+        class Proto(DatasetInfoABC):
+            def __init__(self):
+                self.inputs = "a"
+                self.label_names = "a"
+
+        # intentionally created to check error
+        Proto()  # skipcq
+
+
+def test_KatherDataset(tmp_path):
```

Comment:
I think purest style is to not use capitals in the function title and convert to camel case e.g. `test_kather_dataset`